src/page-objects/InternetIdentityPage.ts

```diff
@@ -153,4 +153,96 @@ export class InternetIdentityPage {
     await iiPage.waitForEvent('close');
     expect(iiPage.isClosed()).toBe(true);
   };
+
+  createNewIdentity = async (params?: {
+    selector?: string;
+    captcha?: boolean;
+  }): Promise<number> => {
+    const iiPagePromise = this.context.waitForEvent('page');
+
+    await this.page.locator(params?.selector ?? '[data-tid=login-button]').click();
+
+    const iiPage = await iiPagePromise;
+    await expect(iiPage).toHaveTitle('Internet Identity');
+
+    await iiPage.locator('#registerButton').click();
+    await iiPage.locator('[data-action=construct-identity]').click();
+
+    if (params?.captcha === true) {
+      await iiPage.locator('input#captchaInput').fill('a', {timeout: 10000});
+      await iiPage.locator('#confirmRegisterButton').click();
+    }
+
+    const identityText = await iiPage.locator('#userNumber').textContent();
+    const createdIdentity = parseInt(identityText!);
+    expect(createdIdentity).not.toBeNull();
+    return createdIdentity;
+  }
+
+  manuallySignInWithIdentity = async ({
+    selector,
+    identity
+  }: {
+    selector?: string;
+    identity: number;
+  }): Promise<void> => {
+    const iiPagePromise = this.context.waitForEvent('page');
+
+    await this.page.locator(selector ?? '[data-tid=login-button]').click();
+
+    const iiPage = await iiPagePromise;
+    await expect(iiPage).toHaveTitle('Internet Identity');
+
+    const existingIdentityLoginLocator = iiPage.locator(`[data-anchor-id="${identity}"]`);
+    const firstRunManualLoginLocator = iiPage.locator('#loginButton');
+    const moreOptionsLoginLocator = iiPage.locator('[data-role="more-options"]');
+
+    const waitOptions: { state: 'visible'; timeout: number } = { state: 'visible', timeout: 10000 };
+
+    type RaceResult = 'existingIdentityLogin' | 'firstRunManualLogin' | 'moreOptionsLogin' | 'none';
+
+    async function getRaceResult(
```

Comment:
nitpick: for consistency with our codebase I would use a `const getRaceResult = () ...`